R/docvars.R

```diff
@@ -75,28 +80,26 @@ make_docvars <- function(n, docname = NULL, unique = TRUE) {
     if (n == 0) {
         data.frame("docname_" = character(),
                    "docid_" = factor(),
-                   "docnum_" = integer(), 
-                   "segnum_" = integer(), 
+                   "segid_" = integer(), 
                    stringsAsFactors = FALSE)
     } else {
         if (unique) {
             docnum <- match(docname, unique(docname))
             if (any(duplicated(docname))) {
-                segnum <- stats::ave(docname == docname, docname, FUN = cumsum)
-                docid <- paste0(docname, ".", segnum)
+                segid <- stats::ave(docname == docname, docname, FUN = cumsum)
+                docid <- paste0(docname, ".", segid)
```

Comment:
This means that not just duplicated docnames, but also all docnames, get a numerical suffix. Is this what we want?
```r
corpus(letters[1:3], docnames = c("A", "A", "B"), unique_docnames = FALSE) %>%
    docnames()
## [1] "A.1" "A.2" "B.1"
```

Would this not be better?
```r
## [1] "A.1" "A.2" "B"
```